packages/wrangler/src/index.tsx

```diff
@@ -168,16 +169,32 @@ function getEntry(config: Config, command: string, script?: string): Entry {
     file = path.resolve(script);
   } else {
     // If the script name comes from the config, then it is relative to the wrangler.toml file.
-    if (config.build?.upload?.main === undefined) {
+    if (config.main === undefined && config.build?.upload?.main === undefined) {
       throw new Error(
-        `Missing entry-point: The entry-point should be specified via the command line (e.g. \`wrangler ${command} path/to/script\`) or the \`build.upload.main\` config field.`
+        `Missing entry-point: The entry-point should be specified via the command line (e.g. \`wrangler ${command} path/to/script\`) or the \`main\` config field.`
       );
+    } else if (
+      config.main === undefined &&
```

Comment:
So we only show the deprecation warning if there is not `main` field...
I think it would be good to do so anyway, since you don't want to have both in the config, right?

Reply:
Good point. In fact, if both are defined, we should probably throw an error.